go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/openHPI/poseidon
 
-go 1.21.1
+go 1.22
```

Comment:
I was trying to identify why I switched to `1.21.1` in a previous commit. This was somewhat related to some tool. If I extracted that correctly from my browser's history, I was motivated by the following comment:

```
https://github.com/golang/go/issues/62278#issuecomment-1693538776
```

Has that behavior changed? If I read the comments in the aforementioned issue correctly, that's not the case. However, I cannot reproduce the issue locally any longer (except by running `GOTOOLCHAIN="go1.22+auto" go mod tidy`, which I didn't do before -- I only executed `go mod tidy` or actually `make tidy-deps`).

Reply:
I see, I wasn't aware of this. Now, I changed the version to `1.22.0`. That is also the fix for most of the linked pull requests.

Reply:
Don't worry, I wasn't aware of that either and really had to focus on finding some plausible explanation for my change 🤓. Obviously, the commit message wasn't that great, back then 🙈 

Thanks for changing. Now, we also need to be aware of this specific version and increase it more often (as minor Go releases get published).